### ISSUE TITLE
Rule vue/component-api-style accepts `<script>` when set to `['error', ['script-setup']]`

### ISSUE THREAD

**Issue by jpolo** [vuejs/eslint-plugin-vue#2056]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.29.0
- **eslint-plugin-vue version:** 9.8.0
- **Node version:** 18.12.1
- **Operating System:**  MacOS

**Please show your full configuration:**

.eslintrc inherits a shared configuration available on the same repository as the test files/ rules, etc. See below for further details.

**What did you do?**	

https://github.com/Captive-Studio/es-project-config/blob/40c66d56bd357c4f7b61bd3512c3315431a48fb8/packages/eslint-config/rules/vue.js#L6
```js
module.exports = concatESConfig({
  extends: ['plugin:vue/vue3-recommended'],
  rules: {
    'vue/component-api-style': [error, ['script-setup']], // Use script-setup as default syntax, can be overriden in project if needed
  },
});
```

**What did you expect to happen?**
https://github.com/Captive-Studio/es-project-config/blob/40c66d56bd357c4f7b61bd3512c3315431a48fb8/packages/eslint-config/_tests/vue-project/Test.vue#L9

Should not pass tests as it is using `<script>` instead of `<script setup>`

**What actually happened?**

Tests are passing which implies vue plugin does not validate correctly (according to documentation)

**Repository to reproduce this issue**

https://github.com/Captive-Studio/es-project-config/blob/40c66d56bd357c4f7b61bd3512c3315431a48fb8/packages/eslint-config

Thanks in advance for your help !

**Comment by ota-meshi** [comment]
`<script setup>` can combine `<script>` with a `default export`, and some options are allowed. The name option is one of them.
https://vuejs.org/api/sfc-script-setup.html#usage-alongside-normal-script

In other words, your *.vue is equivalent to an empty `<script setup>` component and can be considered using `<script setup>`. So I think the rule is working fine.

```vue
<template>
  <div id="myApplication">
    <h1>Hello World</h1>
    <p>little text behind the title</p>
  </div>
</template>

<script>
export default {
  name: 'App',
};
</script>

<style>
#app {
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
```
